util.go

```diff
@@ -44,6 +45,25 @@ func MustGetUUID4String() string {
 	return u.String()
 }
 
+func IsValidPartialUUID4String(str string) bool {
```

Comment:
I would add a **util_test.go** file, and a table-driven unit test for this function. Just a few success/failure cases would do.